graveler/committed/merge_iterator.go

```diff
@@ -91,6 +98,7 @@ func (d *mergeIterator) Next() bool {
 			// added on theirs, but not on ours - continue
 		}
 	}
+	d.err = d.diffIt.Err()
```

Comment:
The test fails because your change reveals a bug - that I forgot to return false after line 96, when the conflict is found.
Can you please add it as well?